Sim_To_DuT_Interface/DuTLogger/DuTLogger.cpp

```diff
@@ -0,0 +1,339 @@
+//
+// Created by marco on 26.10.21.
+//
+
+#include "DuTLogger.h"
+
+// initialize the logging paths
+std::string DuTLogger::currentLogpathConsole = initializeLoggingPath(LOGGER_TYPE::CONSOLE);
+std::string DuTLogger::currentLogpathData = initializeLoggingPath(LOGGER_TYPE::DATA);
+
+// initialize the static handlers
+quill::Handler* DuTLogger::consoleHandler = DuTLogger::buildConsoleHandler();
+quill::Handler* DuTLogger::consoleFileHandler = DuTLogger::buildFileHandler();
+
+// initialize the static loggers
+quill::Logger* DuTLogger::consoleLogger = DuTLogger::createConsoleLogger("consoleLog", false);
+quill::Logger* DuTLogger::consoleFileLogger = DuTLogger::createConsoleLogger("consoleFileLog", true);
+quill::Logger* DuTLogger::dataLogger = DuTLogger::createDataLogger();
+
+
+/**
+ * Starts the quill engine. Won't start it will the engine is already running.
+ */
+void DuTLogger::startEngine() {
+    // initialize everything if it hasn't been done yet
+    if (!startedQuillEngine) {
+        // start the quill engine
+        quill::start();
+
+        // remember that we started the engine and checked everything
+        startedQuillEngine = true;
+    }
+}
+
+/**
+ * Creates a handler to write messages in the console.
+ * This handler can be used to connect it to a logger.
+ *
+ * @return a hadler to log messages in the console
+ */
+quill::Handler* DuTLogger::buildConsoleHandler() {
+    // build a handler for the console
+    quill::Handler* newHandler = consoleHandler = quill::stdout_handler("consoleHandler");
+    newHandler->set_log_level(DEFAULT_CONSOLE_LOG_LEVEL);
+
+    // modify the pattern for the logger
+    newHandler->set_pattern(QUILL_STRING("%(ascii_time)  %(level_name): %(message)"),
+                                "%D %H:%M:%S.%Qms");
+    return newHandler;
+}
+
+/**
+ * Creates a handler to write messages in a file.
+ * This handler can be used to connect it to a logger.
+ * It will create a file for this day, if it hasn't been done yet.
+ *
+ * @return a hadler to log messages in a file
+ */
+quill::Handler* DuTLogger::buildFileHandler() {
+    // a second handler for the file is needed
+    std::string basicPath = currentLogpathConsole + "/Logfile.log";
+    quill::Handler* newHandler = quill::file_handler(basicPath, FILE_MODE_CONSOLE,
+                                                     quill::FilenameAppend::Date);
+    newHandler->set_log_level(DEFAULT_FILE_LOG_LEVEL);
+
+    // modify the pattern for the logger
+    newHandler->set_pattern(QUILL_STRING("%(ascii_time)  %(level_name): %(message)"),
+                            "%D %H:%M:%S.%Qms");
+    // return the new handler
+    return newHandler;
+}
+
+/**
+ * Builds a logger for the console. If necessary the logger will be build with an additional handler for
+ * files. This will be configured by the second argument.
+ *
+ * @param name the name for the logger
```

Comment:
Suggestion: You could format the @param comments to be a bit more readable in the future. 
1. Align the descriptions
2. Start the description with a capital letter